platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/pax/web/jetty/FilterInjector.java

```diff
@@ -0,0 +1,169 @@
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.pax.web.jetty;
+
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+import javax.servlet.DispatcherType;
+import javax.servlet.FilterRegistration;
+import javax.servlet.ServletContext;
+import javax.servlet.SessionCookieConfig;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.hooks.service.EventListenerHook;
+import org.osgi.framework.hooks.service.ListenerHook.ListenerInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Injects the {@link SecurityJavaSubjectFilter} into the {@link ServletContext} with properties
+ * filterName = {@link FilterInjector#SECURITY_JAVA_SUBJECT_FILTER}, urlPatterns = {@link
+ * FilterInjector#ALL_URLS}, and no servlet name as the {@link ServletContext} is being created.
+ * Sets up all {@link javax.servlet.Servlet}s with the {@link SecurityJavaSubjectFilter} in the
+ * {@link javax.servlet.FilterChain}.
+ */
+public class FilterInjector implements EventListenerHook {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FilterInjector.class);
+
+  private static final String ALL_URLS = "/*";
+
+  private static final String SECURITY_JAVA_SUBJECT_FILTER = "security-java-subject-filter";
+
+  private final SecurityJavaSubjectFilter securityJavaSubjectFilter;
+
+  private final ScheduledExecutorService executorService;
+
+  /**
+   * Creates a new filter injector with the specified {@link SecurityJavaSubjectFilter}.
+   *
+   * @param securityJavaSubjectFilter filter that should be injected.
+   * @param executorService used to check for missed servlet contexts
+   */
+  public FilterInjector(
+      SecurityJavaSubjectFilter securityJavaSubjectFilter,
+      ScheduledExecutorService executorService) {
+    this.securityJavaSubjectFilter = securityJavaSubjectFilter;
+    this.executorService = executorService;
+  }
+
+  @Nullable
+  protected BundleContext getContext() {
+    final Bundle cxfBundle = FrameworkUtil.getBundle(SecurityJavaSubjectFilter.class);
+    if (cxfBundle != null) {
```

Comment:
✏️ For simplicity `return cxfBunlde != null ? cxfBundle.getBundleContext() : null;`